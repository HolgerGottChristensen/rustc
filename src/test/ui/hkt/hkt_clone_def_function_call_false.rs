// HKT check that if I is not clone, that we report a not clone error, function call syntax, definition only
// check-fail

fn test<I<%J>>(input: I<u32>) -> I<u32> {
    Clone::clone(&input) //~ ERROR the trait bound `I<u32>: Clone` is not satisfied [E0277]
}

fn main() {}
