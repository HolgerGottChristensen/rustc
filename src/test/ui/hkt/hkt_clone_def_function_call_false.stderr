error[E0277]: the trait bound `I<u32>: Clone` is not satisfied
  --> $DIR/hkt_clone_def_function_call_false.rs:5:18
   |
LL |     Clone::clone(&input)
   |     ------------ ^^^^^^ the trait `Clone` is not implemented for `I<u32>`
   |     |
   |     required by a bound introduced by this call
   |
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
   |
LL | fn test<I<%J>>(input: I<u32>) -> I<u32> where I<u32>: Clone {
   |                                         +++++++++++++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
